doc/architecture.md

```diff
@@ -1,109 +1,176 @@
 
 # **WIP: Under Construction and not yet complete.**
 
-The cognitive architecture is the heart of OnAIR and consists of several components.
+The cognitive architecture is the backbone of OnAIR. The configuration files provide the structure that OnAIR is to use at runtime. They set up the data source, the conduit for telemetry to enter the system, and the pipeline, the series of cognitive components that run.
 
 - [Configuration Files](#the-configuration-files)
   - [Initialization File](#initialization-ini-file)
   - [Telemetry Definitions File](#telemetry-definitions-json-file)
 - [OnAirDataSource](#the-onairdatasource)
+  - [CSV Parser](#csv-parser)
+  - [Redis Adapter](#redis-adapter)
+  - [SBN Adapter](#sbn-adapter)
 - [Pipeline](#the-pipeline)
   - [Data Types](#data-types)
   - [Plugins](#the-plugins)
+  - [Flow](#flow)
 
 ## The Configuration Files
 These files define the components and data structure that will be used by a running instance of OnAIR.
 
 ### Initialization (.ini) File
-Provides the details about the components and other files used by OnAIR to perform its data reasoning.
+Provides the details about the components and other files used by OnAIR when performing its data reasoning.
+
+>#### **[FILES]** (Required)
+>Pathing and file name information  
+>|Key|Required?|Expected Type|Description|
+>|-|-|-|-|
+>|TelemetryFilePath|Yes|str|location of the telemetry file on the local system|
```

Comment:
Should TelemetryFilePath and TelemetryFile not be required for live data sources? Related issue: https://github.com/nasa/OnAIR/issues/107

Reply:
It should not, but currently _is_

Reply:
To clairfy: this documentation is stating what is the case at present. Both TelemetryFilePath and TelemetryFile are required keywords. You can leave them blank when your "adapter" source does not use it. Making them blank in the Redis example .ini (they are not currently blank, fyi) works fine because the adapter even states at start:
```
$ python driver.py onair/config/redis_example.ini
Redis Adapter ignoring file
```
However, you cannot remove the keywords from the .ini file:
```
KeyError: "Config file: 'onair/config/redis_example.ini', missing key: TelemetryFilePath"
```
Therefore, they are still _required_. It would be best to change this, but that is best put into another issue.